starboard/extension/setstring_test.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/client_porting/wrap_main/wrap_main.h"
+#include "starboard/event.h"
+#include "starboard/extension/crash_handler.h"
+#include "starboard/system.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+const char* long_string =
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "LongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLongLong"
+    "String";
+
+TEST(ExtensionTest, SetString) {
+  typedef CobaltExtensionCrashHandlerApi ExtensionApi;
+  const char* kExtensionName = kCobaltExtensionCrashHandlerName;
+
+  const ExtensionApi* extension_api =
+      static_cast<const ExtensionApi*>(SbSystemGetExtension(kExtensionName));
+  if (!extension_api) {
+    return;
+  }
+
+  extension_api->SetString("Annotation1", long_string);
+  extension_api->SetString("Annotation2", long_string);
+  extension_api->SetString("Annotation3", long_string);
+  extension_api->SetString("Annotation4", long_string);
+
+  SbSystemBreakIntoDebugger();
```

Comment:
Do we simply expect this test to crash ? 

Note that our current version of GoogleTest on Starboard does not yet support Death Tests:

https://github.com/google/googletest/blob/main/docs/advanced.md#death-tests

Reply:
Yes, it just crashes to generate a coredump. Would be nice to implement it as a Death Test.

Reply:
Resolved in [73819a1](https://github.com/youtube/cobalt/pull/4443/commits/73819a16a549c874555efdd540982f04d16dfa8c).